src/black/__init__.py

```diff
@@ -625,6 +625,9 @@ def get_sources(
     sources: Set[Path] = set()
     root = ctx.obj["root"]
 
+    _exclude = exclude
```

Comment:
Not sure how to fix the mypy check:

```
src/black/__init__.py:681:21: error: Argument 4 to "gen_python_files" has incompatible type "Optional[Pattern[str]]"; expected "Pattern[str]"  [arg-type]
```

The complain comes because `exclude` has type `Optional[Pattern[str]]`, so `_exclude` should have the same type. But `gen_python_files` always expects a `Pattern[str]`.

In the case that `exclude` is `None` (and therefore `_exclude` too), the value of `_exclude` then changes to `Pattern[str]`, so the value passed to `gen_python_files` will never be `None`. Any ideas?

Reply:
Maybe use[ `typing.cast`](https://docs.python.org/3/library/typing.html#typing.cast) when passing the value to `gen_python_files`? `cast(Pattern[str], _exclude)`. Feels bad but should make mypy happy

Reply:
Decided to add `cast(Pattern[str], _exclude)` when passing the value to `gen_python_files`